src/libraries/Common/src/System/Net/SocketAddress.cs

```diff
@@ -64,23 +69,27 @@ public int Size
                 {
                     throw new IndexOutOfRangeException();
                 }
-                return Buffer[offset];
+                return InternalBuffer[offset];
             }
             set
             {
                 if (offset < 0 || offset >= Size)
```

Comment:
As long as you're making tweaks here, you can make this:
```C#
if ((uint)offset >= (uint)Size)
{
    throw new IndexOutOfRangeException();
}
```
and save one comparison.  Same in the getter.